livre/lib/cache/HistogramCache.cpp

```diff
@@ -0,0 +1,68 @@
+/* Copyright (c) 2011-2015, EPFL/Blue Brain Project
+ *                     Ahmet Bilgili <ahmet.bilgili@epfl.ch>
+ *
+ * This file is part of Livre <https://github.com/BlueBrain/Livre>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <livre/lib/cache/HistogramCache.h>
+#include <livre/lib/cache/HistogramObject.h>
+
+#include <livre/lib/cache/TextureDataCache.h>
+#include <livre/lib/cache/TextureDataObject.h>
+
+namespace livre
+{
+
+struct HistogramCache::Impl
+{
+public:
+    Impl( HistogramCache& histogramCache,
+          const TextureDataCache& dataCache )
+        : _histCache( histogramCache )
+        , _dataCache( dataCache )
+    {}
+
+    CacheObject* generate( const CacheId& cacheId )
+    {
+        return new HistogramObject( cacheId, _histCache );
+    }
+
+   HistogramCache& _histCache;
+   const TextureDataCache& _dataCache;
+};
+
+HistogramCache::HistogramCache( const size_t maxMemBytes,
+                                const TextureDataCache& dataCache )
+    : Cache( "Histogram Cache", maxMemBytes )
+    , _impl( new Impl( *this, dataCache ))
+{}
+
+CacheObject* HistogramCache::_generate( const CacheId& cacheId )
+{
+    return _impl->generate( cacheId );
```

Comment:
Separate function in the impl not needed, simply return new HistogramObject( cacheId, _imple->_histCache );